pex/cli/commands/lock.py

```diff
@@ -99,8 +99,13 @@ def _add_create_arguments(cls, create_parser):
                 "a lock file that contains exactly the distributions that would be used in a local "
                 "resolve. If an sdist would be used, the sdist is included, but if a wheel would "
                 "be used, an accompanying sdist will not be included. The {sources} style includes "
-                "locks containing wheels and the associated sdists when available.".format(
-                    strict=LockStyle.STRICT, sources=LockStyle.SOURCES
+                "locks containing both wheels and the associated sdists when available. The "
+                "{universal} style generates a universal lock for all possible target interpreters "
+                "and platforms, although the scope can be constrained via one or more "
+                "--interpreter-constraint.".format(
```

Comment:
Well, someone else needs to give this guidance I think - you won't like my opinion. My opinion is that strict is preferred - its the only true lock. When you admit sources in addition you admit untested paths. So you've locked, but you have no clue if:
1. The locked sdist, when built a year later, even though it matches the fingerprint at lock time, doesn't unleash havoc with malicious code that does not match the wheel contents.
2. The locked sdist has different Requires-Dist / Requires-Python. This is much better than 1 - you won't suffer an attack, but you will suffer a resolve failure since there will be dependencies missing from the lock.

When you go further and admit universal, its the same problem en-masse.

If you want me to write down all this and are OK with the opinion, I can do that - but none of this is specific to Pex or even Python, this is true conceptually for any locking system where there is more than 1 distribution.

Reply:
I do think your opinion would be worth sharing, John. I suspect not every Pex user will realize this tradeoff, many still are learning about supply chain attacks in the first place.

For example, something like:

> It is often necessary to use {strict} or {universal} for the lock to work in other environments, such as a lock generated on macOS to work on Linux. However, this extra compatibility means that ...

Reply:
Alright then. Will do.

I truly think anyone enabling locks and not thinking about what they're doing is being very American and not very Japanese. There should be more thoughtfulness, deliberation, learning and craft instead of check a box slap-dash get-er-done, but it is true that people don't seem to self-educate that well.

Reply:
> but it is true that people don't seem to self-educate that well.

I view this feature and this soon-to-be help message as part of how we get people on the path to learning. We get them to think about something they may have never considered in the first place because they don't know what questions to ask. By using Pex/Pants, you become a better engineer and are exposed to important practices.

As discussed before, I don't think we need to / want to explain everything in-depth. Instead, it can be particularly compelling to introduce a topic and then link to more in-depth things for people to keep learning.